lib/index.js

```diff
@@ -52,6 +57,14 @@ function getText(fieldBoost, doc) {
   return text;
 }
 
+function handleNestedObjectArrayItem(fieldBoost, deepField) {
+  return function (one) {
+    return getText(utils.extend({}, fieldBoost, {
+      deepField: deepField
+    }), one);
+  };
```

Comment:
He wrote this in a functional style where the function returns a customized function to feed into the map call. That way it maps back to your getText() function and any logic it has in it. It's not my style of coding and I don't have much insight on it but it works...